### ISSUE TITLE
Strip EXIF data or not? (was: EXIF orientation)

### ISSUE THREAD

**Issue by Pinpickle** [bolt/bolt#2363]
I'm not sure what the best way to get around this, or the state of web browsers in honouring EXIF is but the issue presents itself when creating a thumbnail of an image that has EXIF orientation in it (say, from an iPhone) - the thumbnail appears rotated.

For some editors, this seemingly random rotation will be a cause of frustration.

I have a fear that if we change bolt-thumbs to honour EXIF orientation, browsers will still not honour the original image (though this is an issue that probably should still be fixed). I would feel much safer if images were stripped of their orientation and then rotated/mirrored upon upload (keeping all of the other EXIF data in tact). I can't think of a case where this wouldn't be desirable, but it should probably be an option whether it will do this or not.

**Comment by bobdenotter** [comment]
The original bug (rotation of thumbs) was fixed in #2370 and https://github.com/bolt/bolt-thumbs/pull/8. Leaving this open, to see if we should do anything with Exif data or not.

**Comment by cdowdy** [comment]
> Leaving this open, to see if we should do anything with Exif data or not.

I know this is a 20+ day old RFC but I thought I'd throw my two cents in ...  


I agree with this from @bobdenotter 
> As for "keeping" EXIF data.. If we have the chance, I'd rather we err on the safe side, and Bolt should strip it all, unless specifically told not to   
 
From a web performance standpoint ( not art direction like mentioned above for photography site ) stripping all the unnecessary EXIF data can be / is beneficial.  
* [Cloudinary - Mistakes In website Image Handling](http://cloudinary.com/blog/top_10_mistakes_in_handling_website_images_and_how_to_solve_them)  
  * In the majority of the cases, it would be an excellent idea to strip this meta-data off. This is good both for privacy and for reduced file size.  
* [Addy Osmani - Tools for image Optimzation](http://addyosmani.com/blog/image-optimization-tools/)  
  * You may also find that removing EXIF data and unneeded color profile information from images also leads to some gains  

as for honoring EXIF orientation I believe Safari is the only browser that supports that ( with all things web they quickly change so this may no longer be the case ). So iPhones and in turn any browser on the iphone (since they all use the same underlying engine). There is also a CSS3 property to honor EXIF orientation but [browser support](http://caniuse.com/#feat=css-image-orientation) is tuuuuuuuuuuurible.  

Art Directed cases like a photography site have different requirements though

**Comment by Pinpickle** [comment]
I'll chime in on this again.

I think most images that we're using in a template should be thumbnail'd versions to restrict too large a file being downloaded, and thumbnails currently honour EXIF orientation anyway. I just feel iffy about tarnishing the original JPEG by changing quality settings and adding more compression noise.

TLDR; thumbnail everything

**Comment by Intendit** [comment]
I know this is an old thread, but I'll chime in.

One of the original issues were that the rotation would seem random, but after #3308 both the rotated aspect ratio and orientation can be found in the imageinfo() tag. I can also imagine having a more complete exif in that tag with things such as geotagging data and timestamps but that seems like a different question. As for stripping the exif I would say that in 99% of cases that is wanted, and one could still link to the original without using the image() or thumbnail() tags.

If I am correct, right now bolt-thumbs honors the exif orientation (rotates the image), we give the template the orientation and aspect ratio as it will be after the rotation, and bolt-thumbs strips the exif.

**Comment by bobdenotter** [comment]
Agreed with @Intendit. Also, thumbs scrubs the exif tags, so my initial worries are gone too. Closing this one.
